### PR TITLE
feat: output process

### PR DESCRIPTION
**Pull request by micache**
output process with unit test by @codemaivanngu

### CODE REVIEW

File: tests/test_ocr_output.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+import sys
+import os
+
+project_root = os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))
+sys.path.insert(0, project_root)
+FILE_PATH = os.path.join(project_root, 'results')
+
+print(FILE_PATH)
+if not os.path.exists(FILE_PATH):
+    print("jhdfbhjsdafgjkkkkjhasfdggfjdshgjhsfghjsfgdhjgfsdhjgsfdjhgshjfdgsfd")
```
Comment (Huy1711) [inline tests/test_ocr_output.py]: please delete this line

File: .gitignore [hunk 2]
```diff
@@ -160,3 +160,6 @@ cython_debug/
 #  and can be added to the global gitignore or merged into this file.  For a more nuclear
 #  option (not recommended) you can uncomment the following to ignore the entire idea folder.
 #.idea/
+
+results/
+**/miniconda.exe
```
Comment (Huy1711) [inline .gitignore]: What this line do?
If you use conda please put the conda env folder in another folder (not in the repository folder)

File: setup.cfg
Comment (Huy1711) [inline setup.cfg:1]: Please delete this file as we already setup package using setup.py

File: pyproject.toml
Comment (Huy1711) [inline pyproject.toml:1]: Please delete this file as we already setup package using setup.py

File: requirements.txt [hunk 3]
```diff
@@ -1,4 +1,7 @@
 numpy
+pillow
```
Comment (Huy1711) [inline requirements.txt:2]: Duplicate libs, please remove this line

File: src/ocr_app/data_process/client_secrets.json
Comment (Huy1711) [inline src/ocr_app/data_process/client_secrets.json:1]: This is a sensitive file as it contains ggdrive key, please delete this file.

You can store key in a '.env' file instead

File: src/ocr_app/data_process/drive_upload.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
+import os
+
+from pydrive.auth import GoogleAuth
+from pydrive.drive import GoogleDrive
+
+folder_id = '1dGRj7HaiKj2YZNcYZA3MB55ZhrcoXaYM'
```
Comment (Huy1711) [inline src/ocr_app/data_process/drive_upload.py]: Please do not hard code like this, save key in a .env file

File: src/ocr_app/data_process/data_model/ocr_output.py [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+FILE_PATH = os.path.join(
+    os.path.dirname(os.path.dirname(os.path.dirname(os.path.dirname(os.path.dirname(__file__))))),
+    'results',
+)
+print(FILE_PATH)
+if not os.path.exists(FILE_PATH):
+    os.makedirs(FILE_PATH)
+
+from .test import genImg_text
```
Comment (Huy1711) [inline src/ocr_app/data_process/data_model/ocr_output.py]: This is redundant import

File: src/ocr_app/data_process/data_model/ocr_output.py [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+class OcrResults:
+    images: List[np.ndarray]
+    ocr_outputs: List[List[Dict[str, Dict[str, int]]]]
+
+
+imgs, ocrs = genImg_text()
+tmp = OcrResults(imgs, ocrs)
+
+
+class JsonProcessor:
```
Comment (Huy1711) [inline src/ocr_app/data_process/data_model/ocr_output.py]: Please put this class into /src/ocr_app/output_process/json_process.py
Don't forget to create "\_\_init__.py" inside output_process folder

File: src/ocr_app/data_process/data_model/ocr_output.py [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+        return 1234
+
+
+# print(JsonProcessor.process(tmp))
+# JsonProcessor.process(tmp)
+
+from reportlab.lib.utils import ImageReader
+
+
+class OutputImageProcessor:
```
Comment (Huy1711) [inline src/ocr_app/data_process/data_model/ocr_output.py]: Please put this class into /src/ocr_app/output_process/image_process.py

File: src/ocr_app/data_process/data_model/test.py
Comment (Huy1711) [inline src/ocr_app/data_process/data_model/test.py:1]: Please put test file into /tests folder, or if this file is redundant, remove it

File: tests/_test_drive_upload.py [hunk 8]
```diff
@@ -0,0 +1,50 @@
+import os
+import sys
+sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '..', 'src')))
```
Comment (Huy1711) [inline tests/_test_drive_upload.py]: Delete this please

**Review by Huy1711 — COMMENTED** [review]
Please take a look at comments and change the code

File: .gitignore (hunk 2 above)
Comment (micache) [reply .gitignore]: okay i erased

File: src/ocr_app/data_process/client_secrets.json
Comment (Huy1711) [reply src/ocr_app/data_process/client_secrets.json:1]: Up to GGDrive feature is temporarily deleted in this branch

File: src/ocr_app/data_process/drive_upload.py (hunk 4 above)
Comment (Huy1711) [reply src/ocr_app/data_process/drive_upload.py]: Up to GGDrive feature is temporarily deleted in this branch

**Review by Huy1711 — APPROVED** [review]
LGTM

**Review by thaohp113 — APPROVED** [review]

**Pull request merged**